### PR TITLE
Add extra checks before throwing error on ClerkJS initial network calls

### PR DESCRIPTION
**Pull request by anagstef**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [x] `@clerk/shared`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

This PR tries to reduce the errors thrown when ClerkJS tries to initialize and fails on the first `/environment` and `/client` API calls to FAPI.

The following properties of the Navigator API have been used:
- [Navigator.userAgent](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/userAgent), to determine if the page is accessed by web crawlers/bots.
- [Navigator.webdriver](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/webdriver): This flag is enabled by automated tools (e.g. puppeteer) when accessing a web app.
- [Navigator.connection](https://developer.mozilla.org/en-US/docs/Web/API/Navigator/connection): This **experimental** property holds information about the browser's network connection. As it is still an experimental feature, we use it with caution.

If the user trying to initialize ClerkJS, meets one of the above criteria (is offline or bot/automated tool), then the error will be shown as a console warning instead of throwing an actual error.

### CODE REVIEW

File: packages/shared/src/utils/browser.ts
Comment (SokratisVidros) [inline packages/shared/src/utils/browser.ts:5]: 🔧 Please add unit tests for these utility functions.

**Review by SokratisVidros — CHANGES_REQUESTED** [review]
👏 Nice!

File: packages/shared/src/utils/browser.ts
Comment (anagstef) [reply packages/shared/src/utils/browser.ts:5]: @SokratisVidros Sure will do! I was waiting for initial approval.

**Review by dimkl — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]
Nice @anagstef ! Lets add some unit tests and deploy to staging

**Review by desiprisg — APPROVED** [review]

**Review by nikosdouvlis — APPROVED** [review]

**Review by SokratisVidros — APPROVED** [review]

**Pull request merged**
